client/state/selectors/get-site-default-post-format.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * External
+ */
+import { get } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { getRawSite, getSiteOption } from 'state/sites/selectors';
+import { getSiteSettings } from 'state/site-settings/selectors';
+
+/**
+ * Returns the default post format of a site.
+ * Returns null if the site is unknown.
+ *
+ * @param  {Object}    state   Global state tree
+ * @param  {Number}    siteId  The ID of the site we're querying
+ * @return {?String}           The default post format of that site
+ */
+export default function getSiteDefaultPostFormat( state, siteId ) {
+	const site = getRawSite( state, siteId );
+	if ( ! site ) {
+		return null;
+	}
+
+	let defaultPostFormat = get( getSiteSettings( state, siteId ), 'default_post_format' );
+	if ( ! defaultPostFormat ) {
+		defaultPostFormat = getSiteOption( state, siteId, 'default_post_format' );
+	}
+
+	if ( ! defaultPostFormat || defaultPostFormat === '0' ) {
```

Comment:
This is a great catch @aduth, and a perfectly valid scenario. I've updated the logic of the selector to reflect that case too. Here is how it will work now:

* If site is not fetched AND settings are not fetched, will return `null`.
* If site is fetched, but settings are not, use the value from the site.
* If site is not fetched but settings are, use the value from settings.
* If both site and settings are fetched, use the value from settings.
* If the format is `""` or `"0"`, default it to `'standard'`.
* Otherwise, it'll return the actual fetched format.

I've also expanded the unit tests to cover all of these situations. Feel free to have another look @aduth.